.github/actions/fullnode-sync/fullnode_sync.py

```diff
@@ -52,6 +56,30 @@ def get_synced_version_from_index_response(api_index_response, exit_if_none):
   return int(synced_version)
 
 
+def get_metric_from_metrics_port(metric_name):
+  """Gets and returns the metric from the metrics port"""
+  # Ping the metrics port
+  process = subprocess.Popen(["curl", "-s", LOCAL_METRICS_ENDPOINT], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  metrics_response, errors = process.communicate()
+  if metrics_response is None:
+    print_error_and_exit("Exiting! Unable to get the metrics from the localhost. Response is empty.")
+  if errors is not None and errors != b'':
+    print("Found output on stderr for get_synced_version_from_metrics_port: {errors}".format(errors=errors))
+
+  # Parse the metric value
+  try:
+    metrics_response = json.loads(metrics_response)
+    metric_value = metrics_response[metric_name]
+  except Exception as exception:
+    print("Exception caught when getting the metric value: {metric_name}. Exception: {exception}".format(metric_name=metric_name, exception=exception))
+    metric_value = 0
```

Comment:
Why do you need to set the value to be 0 here? This should be `None`, so we error out?

Reply:
No, it's possible that a node doesn't have a value for a metric despite running, i.e., metrics don't default to zero. So, for example, when a node is fast syncing, there's a ~30 second gap between starting up and actually processing the first fast sync chunk to initialize the metric. The same goes for the synced version metric.

So here, we just return `0` as the function is only used to fetch the fast sync metric and the synced version metric, i.e., we don't want to die. We'll keep calling this and if the node doesn't appear to be making progress after the `MAX_TIME_BETWEEN_SYNC_INCREASES_SECS` duration, we'll then kill the node. Does that make sense?

Reply:
I agree that we could make this a lot more generic to handle retries, `None` values at the call sites, etc. But, this is me solving the problem we currently have, not the problem we might have in the future 😄

Reply:
May be add this as a comment for clarity :)

Reply:
Done, thanks Sital. When we return this at some point, I can clean it up.